src/main/java/org/broadinstitute/hellbender/tools/IndexFeatureFile.java

```diff
@@ -84,6 +84,78 @@ protected Object doWork() {
         return indexFile.getAbsolutePath();
     }
 
+    /**
+     * We want to report progress when indexing but the indexing code is in htsjdk and not available to us directly.
+     * The workaround is to make a special decorator 'codec' that gets called on every decoded Feature and this can use used to track progress.
+     * The codec delegates all calls and reports progress as it goes.
+     */
+    private static final class ProgressReportingDelegatingCodec<A extends Feature, B> implements FeatureCodec<A, B>{
+        private final FeatureCodec<A, B> delegatee;
+
+        private final ProgressMeter pm;
+
+        ProgressReportingDelegatingCodec(final FeatureCodec<A, B> delegatee){
+            this.delegatee = delegatee;
+            this.pm = new ProgressMeter();
+        }
+        @Override
+        public Feature decodeLoc(final B b) throws IOException {
+            final Feature f = delegatee.decodeLoc(b);
+            pm.update(f);
+            return f;
+        }
+
+        @Override
+        public A decode(final B b) throws IOException {
+            return delegatee.decode(b);
+        }
+
+        @Override
+        public FeatureCodecHeader readHeader(final B b) throws IOException {
+            return delegatee.readHeader(b);
+        }
+
+        @Override
+        public Class<A> getFeatureType() {
+            return delegatee.getFeatureType();
+        }
+
+        @Override
+        public B makeSourceFromStream(final InputStream bufferedInputStream) {
+            //This is the first call to the codes so we start the progress monitor here.
+            pm.start();
```

Comment:
This doesn't seem 100% reliable in the general case -- recommend starting the progress meter in `decode()/`decodeLoc()` instead, and keep track of whether the meter has already been started.